scripts/split-e2e-tests-codebuild.ts

```diff
@@ -0,0 +1,272 @@
+  'src/__tests__/schema-auth-11.test.ts',
+  'src/__tests__/schema-auth-15.test.ts',
+  'src/__tests__/schema-connection.test.ts',
+  'src/__tests__/transformer-migrations/searchable-migration.test.ts',
+];
+const EXCLUDE_E2E_TESTS = [
+  'src/__tests__/transformer-migrations/searchable-migration.test.ts',
+]
+
+export function loadConfigBase() {
+  return yaml.load(fs.readFileSync(CODEBUILD_CONFIG_BASE_PATH, 'utf8'));
+}
+export function saveConfig(config: any): void {
+  const output = ['# auto generated file. DO NOT EDIT manually', yaml.dump(config, { noRefs: true })];
+  fs.writeFileSync(CODEBUILD_GENERATE_CONFIG_PATH, output.join('\n'));
+}
+export function loadTestTimings(): { timingData: { test: string; medianRuntime: number }[] } {
+  return JSON.parse(fs.readFileSync(TEST_TIMINGS_PATH, 'utf-8'));
+}
+function getTestFiles(dir: string, pattern = 'src/**/*.test.ts'): string[] {
+  return glob.sync(pattern, { cwd: dir });
+}
+type COMPUTE_TYPE = 'BUILD_GENERAL1_MEDIUM' | 'BUILD_GENERAL1_LARGE';
+type BatchBuildJob = {
+  identifier: string;
+  env: {
+    'compute-type': COMPUTE_TYPE;
+    variables: [string: string];
+  };
+};
+type ConfigBase = {
+  batch: {
+    'build-graph': BatchBuildJob[];
+    'fast-fail': boolean;
+  };
+  env: {
+    'compute-type': COMPUTE_TYPE;
+    shell: 'bash';
+    variables: [string: string];
+  };
+};
+const MAX_WORKERS = 4;
+type OS_TYPE = 'w' | 'l';
+type CandidateJob = {
+  region: string;
+  os: OS_TYPE;
+  tests: string[];
+  useParentAccount: boolean;
+};
+const createJob = (os: OS_TYPE, jobIdx: number): CandidateJob => {
+  const region = AWS_REGIONS_TO_RUN_TESTS[jobIdx % AWS_REGIONS_TO_RUN_TESTS.length];
+  return {
+    region,
+    os,
+    tests: [],
+    useParentAccount: false,
+  };
+};
+const getTestNameFromPath = (testSuitePath: string): string => {
+  const startIndex = testSuitePath.lastIndexOf('/') + 1;
+  const endIndex = testSuitePath.lastIndexOf('.test');
+  return testSuitePath
+    .substring(startIndex, endIndex)
+    .split('.e2e')
+    .join('')
+    .split('.')
+    .join('-');
+};
+const splitTests = (
+  baseJobLinux: any,
+  testDirectory: string,
+  isMigration: boolean,
+  pickTests: ((testSuites: string[]) => string[]) | undefined,
+) => {
+  const output: any[] = [];
+  let testSuites = getTestFiles(testDirectory);
+  if (pickTests && typeof pickTests === 'function') {
+    testSuites = pickTests(testSuites);
+  }
+  if (testSuites.length === 0) {
+    return output;
+  }
+  const testFileRunTimes = loadTestTimings().timingData;
+
+  testSuites.sort((a, b) => {
+    const runtimeA = testFileRunTimes.find((t:any) => t.test === a)?.medianRuntime ?? 30;
+    const runtimeB = testFileRunTimes.find((t:any) => t.test === b)?.medianRuntime ?? 30;
+    return runtimeA - runtimeB;
+  });
+  const generateJobsForOS = (os: OS_TYPE) => {
+    const soloJobs = [];
+    let jobIdx = 0;
+    const osJobs = [createJob(os, jobIdx)];
+    jobIdx++;
+    for (let test of testSuites) {
```

Comment:
here we're iterating over tests from lower median time to higher and combining them into batches of size `MAX_WORKERS`. This will result in longer running tests getting batched together increasing the probability to fail.

Reply:
For longer running tests consuming a lot of system resources, they will be filtered out as standalone test (not running in batch with others), which can avoid the failure. Currently there are not enough data points for getting all of these tests. I will figure them out in the post e2e runs